### DISCUSSION TITLE
Loading the trained local and global models in the Prostate Example

### DISCUSSION THREAD

**Discussion by atnassar** [NVIDIA/NVFlare#378] [Q&A] [answered]
Hi,
Thank you for your usual support in resolving the issues while working with the nvflare examples.

Could you please advise how to load the trained local models (at the 4 clients) and the trained global model in the prostate example and test them on a new testing dataset?

I can see there are 2 folders "local_model.pt" and "best_local_model.pt" under the below rout for the MSD client and run_2 (fedavg experiment). Each of these 2 folders includes an "archive" folder that has "data" folder (contains 63 unknown files) and a "data.pkl" file.

examples/prostate/workspace_prostate/client_MSD/run_2/app_client_MSD/

Could you please illustrate further about the content in the 2 aforementioned folders above and how to load the trained models with the updated/best weights?

**Comment by ZiyueXu77** [comment, marked as answer]
@atnassar, thanks for the question! As our other discussion, this question is more about Pytorch/MONAI, rather than NVFlare: you can  load the .pt models into any pytorch script that is able to recognize and use the model (that's why I suggested to write a standalone MONAI trainer). "local_model.pt" is used to track the local model trained on the client, and "best_local_model.pt" is the record for the one with best validation score.  
As for the "archive" folder, to be honest I did not see it in my experiments, @YuanTingHsieh do you have any idea?

**Reply by YuanTingHsieh** [reply]
Hi @atnassar @ZiyueXu77 ,

Since this prostate example is using `torch.save` to save the model checkpoints.
Please use `torch.load` to load the weights back.

For the archive folder (that is inside the ".pt" folder), that is related to how PyTorch internally implements `torch.save` method.
As users we just use `torch.load("local_model.pt")` to load back the weights.

For saving and loading please refer to: https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-model-across-devices

Thanks

**Comment by holgerroth** [comment]
@atnassar, did this answer your question?
